posydon/unit_tests/popsyn/test_norm_pop.py

```diff
@@ -0,0 +1,839 @@
+                                                   M_sim,
+                                                   base_simulation_kwargs,
+                                                   base_simulation_kwargs)
+        assert len(weights) == len(base_pop_data)
+        assert np.all(weights >= 0)
+        assert np.allclose(weights, 1/M_sim)
+        
+    
+    def test_mass_ratio_extension(self,
+                                  base_simulation_kwargs,
+                                  base_population_kwargs,
+                                  base_population_data):
+        '''Sample space for mass ratio extension'''
+        
+        base_pop_data = pop_data(base_simulation_kwargs)
+        M_sim = (base_pop_data['S1_mass_i'].sum()
+                 + base_pop_data['S2_mass_i'].sum())
+        weights = norm_pop.calculate_model_weights(base_pop_data,
+                                                   M_sim,
+                                                   base_simulation_kwargs,
+                                                   base_population_kwargs)
+        assert len(weights) == len(base_pop_data)
+        assert np.all(weights >= 0)
+        np.isclose(base_population_data, weights)
+        
+
+class TestBinaryFractions():
+    
+    def test_single_star_population(self, base_simulation_kwargs):
+        # no binaries
+        base_simulation_kwargs['number_of_binaries'] =  100000
+        base_simulation_kwargs['binary_fraction_const'] = 0
+        base_simulation_kwargs['primary_mass_min'] = 7.
+        base_simulation_kwargs['primary_mass_max'] = 20
+        
+        small_sample = pop_data(base_simulation_kwargs)
+        
+        expanded_kwargs = base_simulation_kwargs.copy()
+        expanded_kwargs['binary_fraction_const'] = 0.7
+        expanded_sample = pop_data(expanded_kwargs)
+        
+        M_sim = (small_sample['S1_mass_i'].sum()
+                 + small_sample['S2_mass_i'].sum())
+        small_weights = norm_pop.calculate_model_weights(small_sample,
+                                                         M_sim,
+                                                         base_simulation_kwargs,
+                                                         expanded_kwargs)
+        
+        M_sim = (expanded_sample['S1_mass_i'].sum()
+                 + expanded_sample['S2_mass_i'].sum())
+        expanded_weights = norm_pop.calculate_model_weights(expanded_sample,
+                                                            M_sim,
+                                                            expanded_kwargs,
+                                                            expanded_kwargs)
+        
+        assert len(expanded_weights)  == len(expanded_sample)
+        
+        # Check just single stars in the population.
+        mask = ((expanded_sample['S1_mass_i'] <= 20)
+                & (expanded_sample['state_i'] == 'initially_single_star'))
```

Comment:
As they use the same mass range, we don't need to filter for the mass:
```suggestion
        mask = (expanded_sample['state_i'] == 'initially_single_star')
```